pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1331,56 +1363,58 @@ impl Timeline {
 
     /// Get cached size of relation if it not updated after specified LSN
     pub fn get_cached_rel_size(&self, tag: &RelTag, lsn: Lsn) -> Option<BlockNumber> {
-        let rel_size_cache = self.rel_size_cache.read().unwrap();
-        if let Some((cached_lsn, nblocks)) = rel_size_cache.map.get(tag) {
+        let rel_size_cache = self.rel_size_primary_cache.read().unwrap();
+        if let Some((cached_lsn, nblocks)) = rel_size_cache.get(tag) {
             if lsn >= *cached_lsn {
                 RELSIZE_CACHE_HITS.inc();
                 return Some(*nblocks);
             }
             RELSIZE_CACHE_MISSES_OLD.inc();
         }
+        let mut rel_size_cache = self.rel_size_replica_cache.lock().unwrap();
+        if let Some(nblock) = rel_size_cache.get(&(lsn, *tag)) {
```

Comment:
Do replicas always make their requests at the same LSN? Doesn't make sense since they are ingesting from SK too. Does this mean they'd always get a miss while ingesting?

Reply:
Static replica always request at the same LSN. For following (hot standby) replica, `request_lsn=replay_lsn`.
If there is active master then in most cases hot-standby replica can use latest cache. But if primary is switched off, then hot standby replica will also request pages at the same LSN (because no one appends WAL).